packages/react/src/search/useInfiniteSearch.ts

```diff
@@ -0,0 +1,263 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { DEFAULT_PAGE_SIZE, useSearchService } from './utils';
+import { GenericFields, SearchParameters } from '@sitecore-content-sdk/search';
+
+/**
+ * Options for the useInfiniteSearch hook.
+ * @public
+ */
+export interface UseInfiniteSearchOptions<T = GenericFields> {
+  /**
+   * The query string to search for.
+   * By default empty string is used.
+   */
+  query?: string;
+  /**
+   * The ID of the search index to use.
+   */
+  searchIndexId: string;
+  /**
+   * Number of results per page.
+   * @default 10
+   */
+  pageSize?: number;
+  /**
+   * Specifies the sorting of the search results.
+   */
+  sort?: SearchParameters<T>['sort'];
+}
+
+/**
+ * The state of the useInfiniteSearch hook.
+ * @public
+ */
+export interface UseInfiniteSearchState<T = GenericFields> {
+  /**
+   * The search results.
+   */
+  results: T[];
+  /**
+   * Whether a search request is currently in progress.
+   */
+  isLoading: boolean;
+  /**
+   * Whether the search request was successful.
+   */
+  isSuccess: boolean;
+  /**
+   * Whether the search request failed.
+   */
+  isError: boolean;
+  /**
+   * Whether a search request for more results is currently in progress.
+   */
+  isLoadingMore: boolean;
+  /**
+   * Whether the search request for more results failed.
+   */
+  isLoadingMoreError: boolean;
+  /**
+   * Load more results.
+   */
+  loadMore: () => void;
+  /**
+   * Whether there are more results available.
+   */
+  hasNextPage: boolean;
+  /**
+   * Total number of results across all pages.
+   */
+  total: number;
+  /**
+   * Total number of pages available based on `total` and `pageSize`.
+   */
+  totalPages: number;
+  /**
+   * The error object if the last search request failed, or null if no error occurred.
+   */
+  error: Error | null;
+}
+
+/**
+ * React hook for performing infinite search queries.
+ * @param {UseInfiniteSearchOptions} options - Configuration options for the infinite search hook.
+ * @returns {UseInfiniteSearchState} The infinite search state.
+ * @public
+ */
+export const useInfiniteSearch = <T = GenericFields>(
```

Comment:
some of the comments from useSearch are also applicable here, please have a look